test/functorch/test_aotdispatch.py

```diff
@@ -2953,29 +2953,105 @@ def forward(self, arg0_1, arg1_1):
+                        y = x.sin()
+                        y.add_(7)
+                        return y.sin()
+
+                    def true_false_fn(x):
+                        return x.cos()
+
+                    return torch.cond(y.cos().shape[0] > 5, true_true_fn, true_false_fn, [y.cos()])
+
+                def false_fn(x):
+                    z = x
+                    z.add_(6)
+                    return z.sin()
+
+                a = torch.cond(x.shape[0] > 4, true_fn, false_fn, [x])
+                return (a + 3, a + 4)
+
+        inp = torch.randn(2, 2)
+        gm, _ = aot_export_module(M(), [inp], trace_joint=False, pre_dispatch=True)
+        self.assertExpectedInline(str(gm.code).strip(), """\
+def forward(self, arg0_1):
+    _set_grad_enabled = torch._C._set_grad_enabled(False)
+    _set_grad_enabled_1 = torch._C._set_grad_enabled(False)
+    _set_grad_enabled_2 = torch._C._set_grad_enabled(False)
+    true_graph_0 = self.true_graph_0
+    false_graph_0 = self.false_graph_0
+    conditional = torch.ops.higher_order.cond(False, true_graph_0, false_graph_0, [arg0_1]);  true_graph_0 = false_graph_0 = arg0_1 = None
+    getitem = conditional[0];  conditional = None
+    add = torch.ops.aten.add.Tensor(getitem, 3)
+    add_1 = torch.ops.aten.add.Tensor(getitem, 4);  getitem = None
+    return (add, add_1)""")  # noqa: B950
+
+        self.assertExpectedInline(str(gm.true_graph_0.code).strip(), """\
+def forward(self, arg0_1):
+    add = torch.ops.aten.add.Tensor(arg0_1, 5);  arg0_1 = None
+    cos = torch.ops.aten.cos.default(add)
+    cos_1 = torch.ops.aten.cos.default(add);  add = None
+    true_graph_0 = self.true_graph_0
+    false_graph_0 = self.false_graph_0
+    conditional = torch.ops.higher_order.cond(False, true_graph_0, false_graph_0, [cos_1]);  true_graph_0 = false_graph_0 = cos_1 = None
+    getitem = conditional[0];  conditional = None
+    return (getitem,)""")  # noqa: B950
+
+        self.assertExpectedInline(str(gm.true_graph_0.true_graph_0.code).strip(), """\
+def forward(self, arg0_1):
+    sin = torch.ops.aten.sin.default(arg0_1);  arg0_1 = None
+    add = torch.ops.aten.add.Tensor(sin, 7);  sin = None
+    sin_1 = torch.ops.aten.sin.default(add);  add = None
+    return (sin_1,)""")
+
     def test_aot_export_predispatch_with_cond(self):
         class M(torch.nn.Module):
             def __init__(self):
                 super().__init__()
-                self.register_buffer("buffer", torch.randn(4, 4))
 
             def forward(self, x):
                 def true_fn(x):
-                    self.buffer.add_(5)
-                    return x.cos() + self.buffer.sum()
+                    y = x
+                    y.add_(5)
+                    return y.cos()
 
                 def false_fn(x):
-                    self.buffer.add_(6)
-                    return x.sin() + self.buffer.sum()
+                    z = x
+                    z.add_(6)
+                    return z.sin()
```

Comment:
These should not trace. cond is supposed to error out if any inputs get mutated.

Reply:
Hmm interesting. I think it still works today.

Reply:
cc @ydwu4

Reply:
yeah, the diff https://github.com/pytorch/pytorch/pull/116799 is not landed. We might want to revitalize it.

Reply:
Ah good catch, thank you. For this PR, I would recommend that we export something that doesn't mutate its inputs then :P